rewrite/rewrite/python/format/tabs_and_indents_visitor.py

```diff
@@ -0,0 +1,397 @@
+            indent += self._style.continuation_indent
+
+        s: Space = self._indent_to(space, indent, loc)
+        if isinstance(cursor_value, J) and not isinstance(cursor_value, EnumValueSet):
+            self.cursor.put_message("last_indent", indent)
+        elif loc == Space.Location.METHOD_SELECT_SUFFIX:
+            raise NotImplementedError("Method select suffix not implemented")
+
+        return s
+
+
+    # NOTE: INCOMPLETE
+    def visit_right_padded(self, right: Optional[JRightPadded[T]],
+                           loc: Union[PyRightPadded.Location, JRightPadded.Location], p: P) -> Optional[
+        JRightPadded[T]]:
+
+        if right is None:
+            return None
+
+        self.cursor = Cursor(self._cursor, right)
+
+        indent: int = cast(int, self.cursor.get_nearest_message("last_indent")) or 0
+
+        t: T = right.element
+        after = right.after
+        # TODO: Check if the visit_and_cast is really required here
+
+        if isinstance(t, J):
+            elem = t
+            if '\n' in right.after.last_whitespace or '\n' in elem.prefix.last_whitespace:
+                if loc in (JRightPadded.Location.FOR_CONDITION,
+                           JRightPadded.Location.FOR_UPDATE):
+                    raise ValueError("This case should not be possible, should be safe for removal...")
+                elif loc in (JRightPadded.Location.METHOD_DECLARATION_PARAMETER,
+                             JRightPadded.Location.RECORD_STATE_VECTOR):
+                    if isinstance(elem, Empty):
+                        # NOTE: DONE
+                        elem = elem.with_prefix(self._indent_to(elem.prefix, indent, loc.after_location))
+                        after = right.after
+                    else:
+                        container: JContainer[J] = cast(JContainer[J], self.cursor.parent_or_throw.value)
+                        elements: List[J] = container.elements
+                        last_arg: J = elements[-1]
+
+                        # TODO: style.MethodDeclarationParameters doesn't exist for Python
+                        # but should be self._style.method_declaration_parameters.align_when_multiple
+                        elem = self.visit_and_cast(elem, J, p)
+                        after = self._indent_to(right.after,
+                                                indent if t is last_arg else self._style.continuation_indent,
+                                                loc.after_location)
+
+                elif loc == JRightPadded.Location.METHOD_INVOCATION_ARGUMENT:
+                    # NOTE: DONE
+                    elem, after = self._visit_method_invocation_argument_j_type(elem, right, indent, loc, p)
+                elif loc in (JRightPadded.Location.NEW_CLASS_ARGUMENTS,
+                             JRightPadded.Location.ARRAY_INDEX,
+                             JRightPadded.Location.PARENTHESES,
+                             JRightPadded.Location.TYPE_PARAMETER):
```

Comment:
I believe this slightly improves the formatting of list literals:
```suggestion
                             JRightPadded.Location.TYPE_PARAMETER,
                             PyRightPadded.Location.COLLECTION_LITERAL_ELEMENT):
```

Reply:
Further, adding `PyRightPadded.Location.DICT_LITERAL_ELEMENT` to this list fixes yet another test.